interfaces/builtin/microstack_support.go

```diff
@@ -0,0 +1,274 @@
+# The path used is hard-coded in libvirt to <huge-page-mnt-dir>/libvirt/qemu.
+/dev/hugepages/libvirt/ rw,
+/dev/hugepages/libvirt/** mrwklix,
+
+# Used by libvirt to read information about available CPU and memory resources of a host.
+/sys/devices/system/cpu/ r,
+/sys/devices/system/node/ r,
+/sys/devices/system/node/node[0-9]*/meminfo r,
+/sys/devices/system/node/node[0-9]*/hugepages/** r,
+
+/sys/module/vhost/parameters/max_mem_regions r,
+
+# Allow nested virtualization checks for different CPU models and architectures (where it is supported).
+/sys/module/kvm_intel/parameters/nested r,
+/sys/module/kvm_amd/parameters/nested r,
+/sys/module/kvm_hv/parameters/nested r, # PPC64.
+/sys/module/kvm/parameters/nested r, # S390.
+
+# Used by libvirt (cgroup-related):
+/sys/fs/cgroup/** rw,
+/sys/fs/cgroup/*/machine/qemu-*/** rw,
+/sys/fs/cgroup/unified/cgroup.controllers r,
+/proc/self/cgroup r,
+/proc/cgroups r,
+
+# Used by libvirt.
+/proc/cmdline r,
+/proc/filesystems r,
+/proc/mtrr w,
+/proc/@{pids}/environ r,
+/proc/@{pids}/mountinfo r,
+/proc/@{pids}/mounts r,
+/proc/@{pids}/sched r,
+/proc/@{pids}/stat r,
+
+# Per man(5) proc, the kernel enforces that a thread may
+# only modify its comm value or those in its thread group.
+owner /proc/@{pid}/task/@{tid}/comm rw,
+
+/proc/*/cmdline r,
+/proc/sys/net/** r,
+/proc/version r,
+/proc/*/stat r,
+/proc/*/mounts r,
+/proc/*/status r,
+/proc/*/ns/net r,
+
+
+# Used by libvirt to work with network devices created for VMs.
+# E.g. getting operational state and speed of tap devices.
+/sys/class/net/tap*/* rw,
+
+# for usb access
+/dev/bus/usb/ r,
+/sys/bus/ r,
+/sys/class/ r,
+# For hostdev access. The actual devices will be added dynamically
+/sys/bus/usb/devices/ r,
+/sys/devices/**/usb[0-9]*/** r,
+# libusb needs udev data about usb devices (~equal to content of lsusb -v)
+/run/udev/data/+usb* r,
+/run/udev/data/c16[6,7]* r,
+/run/udev/data/c18[0,8,9]* r,
+
+# Libvirt needs access to the PCI config space in order to be able to reset devices.
+/sys/devices/pci*/**/config rw,
+
+# for vfio hotplug on systems without static vfio (LP: #1775777)
+/dev/vfio/vfio rw,
+/dev/vfio/* rw,
+
+# Allow querying the max number of segments for a block device by Libvirt
+# (see QEMU commit 9103f1ce and Linux/block/queue-sysfs.txt).
+/sys/devices/**/block/*/queue/max_segments r,
+
+# required by libpmem init to fts_open()/fts_read() the symlinks in
+# /sys/bus/nd/devices
+/ r, # harmless on any lsb compliant system
+/sys/bus/nd/devices/{,**/} r,
+
+# For ppc device-tree access by Libvirt.
+/proc/device-tree/ r,
+/proc/device-tree/** r,
+/sys/firmware/devicetree/** r,
+
+# "virsh console" support
+/dev/pts/* rw,
+# Used by libvirt.
+/dev/ptmx rw,
+# spice
+owner /{dev,run}/shm/spice.* rw,
+
+# Used by libvirt to create lock files for /dev/pts/<num> devices
+# when handling virsh console access requests.
+/run/lock/ r,
+/run/lock/snap.@{SNAP_INSTANCE_NAME}/ rw,
+/run/lock/snap.@{SNAP_INSTANCE_NAME}/** mrwklix,
```

Comment:
This is discussed in https://github.com/snapcore/snapd/pull/8909 where I mention the FHS says all the files for console locking should be `LCK..<something>` so that different processes can coordinate (see https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s09.html). Since putting in snap-specific directories is against the FHS and makes it difficult to actually coordinate, please adjust this to be:

```
/run/lock/ r,
/run/lock/LCK.._pts_* rwk,
```

(tested locally). If `/run/lock/LCK.._pts_*` is not sufficient, please state examples.